geth/peers/peerpool_test.go

```diff
@@ -187,6 +182,13 @@ func (s *PeerPoolSimulationSuite) TestSingleTopicDiscoveryWithFailover() {
 	s.Require().NoError(peerPool.Start(s.peers[1]))
 	defer peerPool.Stop()
 	s.Equal(signal.EventDiscoveryStarted, s.getPoolEvent(poolEvents))
+
+	// create and start topic registry
+	register := NewRegister(topic)
```

Comment:
so it must be registered after topic is advertised or what was the point?

Reply:
I have seen some tests where peer pool was struggling to find the advertised peers. My assumption was that the topic is advertised before the peer pool is started and that may lead to timeouts. I was not able to confirm that locally though.

Reply:
And this assumption was wrong. I got the same issue with this change.